modules/docs/arrow-docs/docs/docs/typeclasses/foldable/README.md

```diff
@@ -9,8 +9,518 @@ permalink: /docs/typeclasses/foldable/
 {:.intermediate}
 intermediate
 
-TODO. Meanwhile you can find a short description in the [intro to typeclasses]({{ '/docs/typeclasses/intro/' | relative_url }}).
+The Typeclass `Foldable` provide us the ability of, given a type `F<A>`, fold their values `<A>`.
 
+`Foldable<F>` is implemented in terms of two basic methods:
+
+- `foldLeft(fa, b)(f)` eagerly folds `fa` from left-to-right.
+- `foldRight(fa, b)(f)` lazily folds `fa` from right-to-left.
+
+Beyond these it provides many other useful methods related to folding over `F<A>` values.
+
+For the following examples we are going to use some common imports 
+
+```kotlin:ank:silent
+import arrow.Kind
+import arrow.core.*
+import arrow.data.ListK
+import arrow.data.k
+import arrow.instances.monoid
+import arrow.instances.listk.foldable.foldable
+import arrow.instances.option.foldable.foldable
+import arrow.typeclasses.Foldable
+```
+ 
+and the same two variables to see the different behaviors of `Foldable`:
+
+```kotlin:ank:silent
+val maybeStr: Option<String> = Some("abc")
+val strList: ListK<String> = listOf("a", "b", "c").k()
+```
+
+### FoldLeft
+Left associative fold on `F` using the provided function.
+
+```kotlin:ank:silent
+fun <F> foldLeft(strKind: Kind<F, String>, FO: Foldable<F>): String =
+  with(FO) {
+    strKind.foldLeft("str: ") { base: String, value: String -> base + value }
+  }
+```
+
+```kotlin:ank
+foldLeft(maybeStr, Option.foldable())
+```
+
+```kotlin:ank
+foldLeft(None, Option.foldable())
+```
+
+```kotlin:ank
+foldLeft(strList, ListK.foldable())
+```
+
+### FoldRight
+Right associative lazy fold on `F` using the provided function.
+
+This method evaluates `lb` lazily, and returns a lazy value.
```

Comment:
Explain that this is because we'd like to avoid StackOverflows for some datatypes.